### PR TITLE
Rework globals

### PR DESCRIPTION
**Pull request by vext01**
This implements an idea Laurie had for locating global symbols.

ykllvm change ~~coming soon~~ (and should go in first): https://github.com/ykjit/ykllvm/pull/122

See commits.

### CODE REVIEW

**Comment by ltratt** [comment]
LGMT.

**Comment by ptersilie** [comment]
In regards to testing this: with https://github.com/ykjit/yk/pull/1030 being merged now we get a normal panic when the guard fails, so it should flush the puts. Have you tried this?

File: tests/c/simple.newcg.c [hunk 1]
```diff
@@ -18,7 +18,7 @@
 //     ...
 //     --- End jit-pre-opt ---
 //     ...
-//     deopt...
```
Comment (ptersilie) [inline tests/c/simple.newcg.c:21]: Could you leave this "deopt" string in here so we know if we break guards and we don't deopt anymore?

Comment (vext01) [reply tests/c/simple.newcg.c:21]: Not in this PR, because the trace doesn't loop.

My next PR brings this back.

**Comment by ltratt** [comment]
@vext01 Have you seen https://github.com/ykjit/yk/pull/1051#issuecomment-2011973596 ?

**Comment by vext01** [comment]
Yes, but the answer is the same as the other comment: the test doesn't make it to deopt, since the trace doesn't loop.

The test is expected to do 2 iterations of the loop in JITted code, but we don't yet have the ability to do >1 iteration. My next PR will fix this.

**Comment by ltratt** [comment]
Got it! Please force push a rebase against a master.

**Comment by vext01** [comment]
rebased.

**Comment by vext01** [comment]
(requires the ykllvm change to merge)

File: tests/c/simple.newcg.c (hunk 1 above)
Comment (ptersilie) [reply tests/c/simple.newcg.c:21]: It shouldn't loop anyway. For me it deopts immediately.

**Comment by ptersilie** [comment]
> Yes, but the answer is the same as the other comment: the test doesn't make it to deopt, since the trace doesn't loop.
> 
> The test is expected to do 2 iterations of the loop in JITted code, but we don't yet have the ability to do >1 iteration. My next PR will fix this.

Why do I get the `deopt` panic then? For me it doesn't loop. Or did this PR introduce changes that now make this loop. In which case, why don't we change it back so it deopts in the first execution of the trace.

**Comment by ltratt** [comment]
I'm interpreting this discussion as something that can easily be changed in a future PR if necessary, so I can merge.

**Pull request merged**